src/jaxsim/high_level/model.py

```diff
@@ -1137,25 +1137,36 @@ def forward_dynamics_crb(
         τ = jnp.atleast_1d(τ.squeeze())
         τ = jnp.vstack(τ) if τ.size > 0 else jnp.empty(shape=(0, 1))
 
+        # Extract motor parameters from the physics model
+        GR = self.motor_gear_ratios()
+        IM = self.motor_inertias()
+        KV = jnp.diag(self.motor_viscous_frictions())
+
+        # Compute auxiliary quantities
+        Γ = jnp.diag(GR)
+        K̅ᵥ = Γ.T @ KV @ Γ
+
         # Compute terms of the floating-base EoM
         M = self.free_floating_mass_matrix()
         h = jnp.vstack(self.free_floating_bias_forces())
         J = self.generalized_free_floating_jacobian()
         f_ext = jnp.vstack(self.external_forces().flatten())
         S = jnp.block([jnp.zeros(shape=(self.dofs(), 6)), jnp.eye(self.dofs())]).T
 
+        # Configure the slice for fixed/floating base robots
+        sl = np.s_[0:] if self.physics_model.is_floating_base else np.s_[6:]
+        sl_m = np.s_[-self.dofs() :]
+
+        # Add the motor related terms to the EoM
+        M = M.at[sl_m, sl_m].set(M[sl_m, sl_m] + jnp.diag(Γ.T @ IM @ Γ))
+        h = h.at[sl_m].set(h[sl_m] + K̅ᵥ @ self.joint_velocities()[:, None])
+        S = S.at[sl_m].set(S[sl_m])
+
         # Compute the generalized acceleration by inverting the EoM
-        ν̇ = jax.lax.select(
-            pred=self.floating_base(),
-            on_true=jnp.linalg.inv(M) @ ((S @ τ) - h + J.T @ f_ext),
-            on_false=jnp.vstack(
-                [
-                    jnp.zeros(shape=(6, 1)),
-                    jnp.linalg.inv(M[6:, 6:])
-                    @ ((S @ τ)[6:] - h[6:] + J[:, 6:].T @ f_ext),
-                ]
-            ),
-        ).squeeze()
```

Comment:
These lines were necessary if I recall to jit-compile this method and get correct results on fixed-based robots. In that case, the generalized acceleration should be computed differently from the floating-base case.